wrangler-core/src/main/java/io/cdap/directives/column/SetHeader.java

```diff
@@ -95,11 +95,17 @@ public Mutation lineage() {
 
   @Override
   public Schema getOutputSchema(SchemaResolutionContext context) {
-    List<Schema.Field> fields = new ArrayList<>(context.getInputSchema().getFields());
+    List<Schema.Field> inputFields = context.getInputSchema().getFields();
+    List<Schema.Field> outputFields = new ArrayList<>();
     for (int i = 0; i < columns.size(); i++) {
-      Schema.Field existing = fields.remove(i);
-      fields.add(i, Schema.Field.of(columns.get(i), existing.getSchema()));
+      if (i < inputFields.size()) {
+        outputFields.add(Schema.Field.of(columns.get(i), inputFields.get(i).getSchema()));
+      }
+    }
+    // Leftover columns (not renamed)
+    for (int i = columns.size(); i < inputFields.size(); i++) {
```

Comment:
from this logic, if `columns` is larger than `inputFields`, the extra columns from `columns` are not added to the output schema. Or am I missing anything?